src/main/java/htsjdk/variant/vcf/VCFHeader.java

```diff
@@ -188,7 +188,10 @@ public void addMetaDataLine(final VCFHeaderLine headerLine) {
      * @return all of the VCF header lines of the ##contig form in order, or an empty list if none were present
      */
     public List<VCFContigHeaderLine> getContigLines() {
-        return Collections.unmodifiableList(contigMetaData);
+        List<VCFContigHeaderLine> contigLines = new ArrayList<>(contigMetaData.size());
+        // this must preserve input order
+        contigLines.addAll(contigMetaData.values());
+        return Collections.unmodifiableList(contigLines);
```

Comment:
Good idea to use the constructor that takes a collection here like @pshapiro4broad suggests though I think.